packages/amplify-category-auth/resources/auth-custom-resource/openIdLambda.js

```diff
@@ -10,57 +10,60 @@ const iam = new IAMClient({});
 
 exports.handler = (event, context) => {
   // Don't return promise, response.send() marks context as done internally
-  const ignoredPromise = handleEvent(event, context);
+  void tryHandleEvent(event, context);
 };
 
-async function handleEvent(event, context) {
+async function tryHandleEvent(event, context) {
   try {
-    if (event.RequestType === 'Update' || event.RequestType === 'Create') {
-      const params = {
-        ClientIDList: event.ResourceProperties.clientIdList.split(','),
-        ThumbprintList: ['0000000000000000000000000000000000000000'],
-        Url: event.ResourceProperties.url,
+    const res = await handleEvent(event);
+    response.send(event, context, response.SUCCESS, res);
+  } catch (e) {
+    response.send(event, context, response.FAILED, { e });
+  }
+}
+
+async function handleEvent(event) {
+  if (event.RequestType === 'Update' || event.RequestType === 'Create') {
```

Comment:
it'd be nice to invert this condition and early return, but I'll leave that up to you and your patience for recording snapshots ;)